projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -79,22 +84,33 @@ export class MerchandisingCarouselComponent {
     return metadata;
   }
 
-  private mapMerchandisingProductsItems(
+  private mapMerchandisingProductsToCarouselItems(
     merchandisingProducts: MerchandisingProducts
-  ): {
-    items$: Observable<Product>[];
-    metadata: Map<string, string>;
-  } {
-    return {
-      items$:
-        merchandisingProducts && merchandisingProducts.products
-          ? merchandisingProducts.products.map(product => of(product))
-          : [EMPTY],
-      metadata:
-        merchandisingProducts && merchandisingProducts.metadata
-          ? merchandisingProducts.metadata
-          : undefined,
-    };
+  ): Observable<MerchandisingProduct>[] {
+    return merchandisingProducts && merchandisingProducts.products
+      ? merchandisingProducts.products.map((product, index) => {
+          product.metadata = this.getCarouselItemMetadata(product, index + 1);
+          return of(product);
+        })
+      : [EMPTY];
+  }
+
+  getCarouselItemMetadata(
```

Comment:
Is this method used in the template? If not, please make it `private`.

Reply:
Nope, I tried but I got "super responsiveness" again 😉 , will make it private

Reply:
Haha, _reactive_ ftw :)